src/rcheevos/alloc.c

```diff
@@ -10,22 +11,43 @@ void* rc_alloc(void* pointer, int* offset, int size, int alignment, rc_scratch_t
   if (pointer != 0) {
     ptr = (void*)((char*)pointer + *offset);
   }
+  else if (scratch != 0) {
+    ptr = &scratch->obj;
+  }
```

Comment:
`rc_alloc_str` calls `rc_alloc` with a NULL `scratch`, so we can't transpose the subobject to return null. Had to add an explicit check